legacy-commands-v1/src/main/java/io/github/minecraftcursedlegacy/mixin/command/MixinMinecraft.java

```diff
@@ -0,0 +1,30 @@
+package io.github.minecraftcursedlegacy.mixin.command;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import io.github.minecraftcursedlegacy.api.command.DispatcherRegistry;
+import net.minecraft.client.Minecraft;
+import net.minecraft.level.Level;
+
+@Mixin(Minecraft.class)
+public class MixinMinecraft {
+	@Redirect(method = "tick()V", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/Minecraft;hasLevel()Z", ordinal = 0))
+	private boolean canOpenChatScreen(Minecraft minecraft) {
+		Level level = ((Minecraft) (Object) this).level;
+		return level != null && (DispatcherRegistry.hasSingleplayerCommands() || level.isClient);
+	}
+
+	@Inject(at = @At("HEAD"), method = "handleClientCommand", cancellable = true)
+	private void handleClientCommand(String command, CallbackInfoReturnable<Boolean> info) {
+		if (command.length() > 1 && command.startsWith("/") && !((Minecraft) (Object) this).isConnectedToServer()) {
```

Comment:
Is it worth checking the length of `command` if you do `startsWith("/")` anyway?

Reply:
It seems you don't see what I am doing. I am checking the command isn't just "/"

Reply:
Maybe I might want to add an empty string command :P